tests/validation/tests/v3_api/test_kdm_changes.py

```diff
@@ -0,0 +1,134 @@
+        aws_nodes = \
+            AmazonWebServices().create_multiple_nodes(
+                node_count, random_test_name(HOST_NAME))
+        i = 0
+        for plug in plugin:
+            if plug == "calico" or plug == "canal" or plug == "weave":
+                rke_config_new["network"]["options"] = \
+                    {"flannel_backend_type": "vxlan"}
+            rke_config_new["network"] = {"type": "networkConfig",
+                                         "plugin": plug}
+            client = get_user_client()
+            cluster_name = random_name()
+            cluster = client.create_cluster(name=cluster_name,
+                                            driver="rancherKubernetesEngine",
+                                            rancherKubernetesEngineConfig=
+                                            rke_config_new)
+            p1 = multiprocessing.Process(target=validate_custom_cluster_kdm,
+                                         args=(cluster,
+                                               rke_config_new,
+                                               aws_nodes
+                                               [i:i+NODE_COUNT_KDM_CLUSTER]))
+            CLUSTER_LIST.append(cluster)
+            list_process.append(p1)
+            p1.start()
+            i = i + NODE_COUNT_KDM_CLUSTER
+    for process in list_process:
+        process.join()
+    env_details = "env.CATTLE_TEST_URL='" + CATTLE_TEST_URL + "'\n"
+    env_details += "env.ADMIN_TOKEN='" + ADMIN_TOKEN + "'\n"
+    env_details += "env.USER_TOKEN='" + USER_TOKEN + "'\n"
+    failed_cluster = {}
+    i = 0
+    for cluster in CLUSTER_LIST:
+        env_details += \
+            "env.CLUSTER_NAME_" + str(i) + "='" + cluster.name + "'\n"
+        i = i + 1
+    create_config_file(env_details)
+    print("env_details:", env_details)
+    for cluster in CLUSTER_LIST:
+        try:
+            validate_cluster(client, cluster, cluster.state, False, False)
+        except:
+            print("Issue in ", cluster.name)
+            failed_cluster[cluster.name]["k8s"] = \
+                cluster["rancherKubernetesEngineConfig"]["kubernetesVersion"]
+            failed_cluster[cluster.name]["network"] = \
+                cluster["rancherKubernetesEngineConfig"]["network"]["plugin"]
+            failed_cluster[cluster.name]["state"] = cluster.state
+    print("--------------Cluster information--------------'\n")
+    print(failed_cluster)
+    assert len(failed_cluster) == 0, "Clusters have failed to provision. " \
+                                     "Check logs for more info"
+
+
+def validate_custom_cluster_kdm(cluster, rke_config_new, aws_nodes):
+    if NODE_COUNT_KDM_CLUSTER == 4:
+        node_roles = [["controlplane"], ["etcd"], ["worker"], ["worker"]]
+    else:
+        node_roles = [["worker", "controlplane", "etcd"]]
+    print("validate function:")
+    print(rke_config_new["network"]["plugin"])
+    print(rke_config_new["kubernetesVersion"])
```

Comment:
Can we have the clusterId/name printed as well ? Also have it printed in the same line  clusterId/name  -  K8s version - plugin . When failures happens , we  only report clusterId/name and we will be able to get to the configuration that failed when clusterId/name gets printed.

Reply:
These print statements were for my testing. The clusters that have succeeded validation will be printed after the statement in the try block.